contracts/utils/ERC7930.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {Bytes} from "@openzeppelin/contracts/utils/Bytes.sol";
+import {Calldata} from "@openzeppelin/contracts/utils/Calldata.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+// 0x00010000010114D8DA6BF26964AF9D7EED9E03E53415D37AA96045
+// 0x000100022045296998a6f8e2a784db5d9f95e18fc23f70441a1039446801089879b08c7ef02005333498d5aea4ae009585c43f7b8c30df8e70187d4a713d134f977fc8dfe0b5
+// 0x000100000014D8DA6BF26964AF9D7EED9E03E53415D37AA96045
+// 0x000100022045296998a6f8e2a784db5d9f95e18fc23f70441a1039446801089879b08c7ef000
+// 0x0001000002A4B114D8DA6BF26964AF9D7EED9E03E53415D37AA96045
+
+library ERC7930 {
+    using SafeCast for uint256;
+    using Bytes for bytes;
+
+    error ERC7930ParsingError(bytes);
```

Comment:
maybe?

```suggestion
    error ERC7930ChainIdTooLarge(bytes);
```

Reply:
the `ERC7930ParsingError` is triggered when calling `parseV1` or `parseV1Calldata` with an input that cannot be decoded. 

That would happen if:
- cannot decode version, chainType, or chainReferenceLength because input buffer is too short
- input version is not 1
- cannot extract chainReference because input buffer is too short
- cannot decode addressLength because input buffer is too short
- cannot extract address because input buffer is too short

So basically "inproper encoding" or "unsupported version".